modules/homebrew.nix

```
@@ -1,219 +1,780 @@
+  });
+
+  mkProcessedSubmodConfig = attrs: mapAttrs (_: mkBrewfileLineValueString)
+    (filterAttrsRecursive (n: v: n != "_module" && n != "brewfileLine" && v != null) attrs);
+
+
+  # Submodules -------------------------------------------------------------------------------------
+  # Option values and descriptions of Brewfile entries are sourced/derived from:
+  #   * `brew` manpage: https://docs.brew.sh/Manpage
+  #   * `brew bundle` source files (at https://github.com/Homebrew/homebrew-bundle/tree/9fffe077f1a5a722ed5bd26a87ed622e8cb64e0c):
+  #     * lib/bundle/dsl.rb
+  #     * lib/bundle/{brew,cask,tap}_installer.rb
+  #     * spec/bundle/{brew,cask,tap}_installer_spec.rb
+
+  onActivationOptions = { config, ... }: {
+    options = {
+      cleanup = mkOption {
+        type = types.enum [ "none" "uninstall" "zap" ];
+        default = "none";
+        example = "uninstall";
+        description = ''
+          This option manages what happens to formulae installed by Homebrew, that aren't present in
+          the Brewfile generated by this module, during <command>nix-darwin</command> system
+          activation.
+
+          When set to <literal>"none"</literal> (the default), formulae not present in the generated
+          Brewfile are left installed.
+
+          When set to <literal>"uninstall"</literal>, <command>nix-darwin</command> invokes
+          <command>brew bundle [install]</command> with the <command>--cleanup</command> flag. This
+          uninstalls all formulae not listed in generated Brewfile, i.e.,
+          <command>brew uninstall</command> is run for those formulae.
+
+          When set to <literal>"zap"</literal>, <command>nix-darwin</command> invokes
+          <command>brew bundle [install]</command> with the <command>--cleanup --zap</command>
+          flags. This uninstalls all formulae not listed in the generated Brewfile, and if the
+          formula is a cask, removes all files associated with that cask. In other words,
+          <command>brew uninstall --zap</command> is run for all those formulae.
+
+          If you plan on exclusively using <command>nix-darwin</command> to manage formulae
+          installed by Homebrew, you probably want to set this option to
+          <literal>"uninstall"</literal> or <literal>"zap"</literal>.
+        '';
+      };
+      autoUpdate = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Homebrew to auto-update itself and all formulae during
+          <command>nix-darwin</command> system activation. The default is <literal>false</literal>
+          so that repeated invocations of <command>darwin-rebuild switch</command> are idempotent.
+
+          Note that Homebrew auto-updates when it's been more then 5 minutes since it last updated.
+
+          If you prefer this behavior you may want to consider enabling
+          ${mkDocOptionLink "homebrew.global.noAutoUpdate"}.
```

```suggestion
          If you do not prefer this behavior you may want to consider disabling
          ${mkDocOptionLink "homebrew.global.autoUpdate"}.
```

I think what we want to say here is to turn the global option off if we never want brew to automatically update, right?

---

Another great catch. Fixed.